src/services/refactors/convertImport.ts

```diff
@@ -0,0 +1,113 @@
+        getAvailableActions(context): ApplicableRefactorInfo[] | undefined {
+            const i = getImportToConvert(context);
+            if (!i) return undefined;
+            const description = i.kind === SyntaxKind.NamespaceImport ? Diagnostics.Convert_namespace_import_to_named_imports.message : Diagnostics.Convert_named_imports_to_namespace_import.message;
+            const actionName = i.kind === SyntaxKind.NamespaceImport ? actionNameNamespaceToNamed : actionNameNamedToNamespace;
+            return [{ name: refactorName, description, actions: [{ name: actionName, description }] }];
+        },
+        getEditsForAction(context, actionName): RefactorEditInfo {
+            Debug.assert(actionName === actionNameNamespaceToNamed || actionName === actionNameNamedToNamespace);
+            const edits = textChanges.ChangeTracker.with(context, t => doChange(context.file, context.program.getTypeChecker(), t, Debug.assertDefined(getImportToConvert(context))));
+            return { edits, renameFilename: undefined, renameLocation: undefined };
+        }
+    });
+
+    // Can convert imports of the form `import * as m from "m";` or `import d, { x, y } from "m";`.
+    function getImportToConvert(context: RefactorContext): NamedImportBindings | undefined {
+        const { file } = context;
+        const span = getRefactorContextSpan(context);
+        const token = getTokenAtPosition(file, span.start, /*includeJsDocComment*/ false);
+        const importDecl = getParentNodeInSpan(token, file, span);
+        if (!importDecl || !isImportDeclaration(importDecl)) return undefined;
+        const { importClause } = importDecl;
+        return importClause && importClause.namedBindings;
+    }
+
+    function doChange(sourceFile: SourceFile, checker: TypeChecker, changes: textChanges.ChangeTracker, toConvert: NamedImportBindings): void {
+        const usedIdentifiers = createMap<true>();
+        forEachFreeIdentifier(sourceFile, id => usedIdentifiers.set(id.text, true));
+
+        if (toConvert.kind === SyntaxKind.NamespaceImport) {
+            doChangeNamespaceToNamed(sourceFile, checker, changes, toConvert, usedIdentifiers);
+        }
+        else {
+            doChangeNamedToNamespace(sourceFile, checker, changes, toConvert, usedIdentifiers);
+        }
+    }
+
+    function doChangeNamespaceToNamed(sourceFile: SourceFile, checker: TypeChecker, changes: textChanges.ChangeTracker, toConvert: NamespaceImport, usedIdentifiers: ReadonlyMap<true>): void {
+        const namespaceSymbol = checker.getSymbolAtLocation(toConvert.name)!;
+
+        // We may need to change `mod.x` to `_x` to avoid a name conflict.
+        const exportNameToImportName = createMap<string>();
+        let usedAsNamespace = false;
+
+        forEachFreeIdentifier(sourceFile, id => {
+            if (id === toConvert.name || checker.getSymbolAtLocation(id) !== namespaceSymbol) return;
+
+            if (!isPropertyAccessExpression(id.parent)) {

```

Comment:
what if it is used in a call expression.. `import * as n from "mod"; n();`? 
if `allowSyntheticDefaultImport` or `esModuleInterop` are on, then you can change it to `import {default as n} from "mod"; n();`, if not, we need to keep the namespace import around.. do not think there is another way to handle that.